OpenRA.Mods.TS/UtilityCommands/LegacyTilesetImporter.cs

```diff
@@ -67,6 +85,8 @@ public void Run(ModData modData, string[] args)
 					var sectionCount = int.Parse(section.GetValue("TilesInSet", "1"));
 					var sectionFilename = section.GetValue("FileName", "").ToLowerInvariant();
 					var sectionCategory = section.GetValue("SetName", "");
+					if (!string.IsNullOrEmpty(sectionCategory) && sectionFilename != "blank")
+						categories.Add(sectionCategory);
```

Comment:
This includes categories for tiles that don't exist (e.g. "MM Ramps" and "Obsolete Height Pieces").  Please move this below the loop, and use a flag that is set inside the loop by the first valid tile to ensure that only valid categories are added to the list.